interfaces/cython/cantera/reactor.pyx

```diff
@@ -859,12 +884,17 @@ cdef class ReactorNet:
         self._reactors.append(r)
         self.net.addReactor(deref(r.reactor))
 
-    def advance(self, double t):
+    def advance(self, double t, pybool apply_limit=True):
```

Comment:
Default value true or false?

Reply:
False. We need to maintain backwards compatibility.

Reply:
A default of `True` is also backwards compatible, though, since by default there aren't any limits to apply.

Reply:
OK, that's fair. When I wrote that comment, I was also thinking about whether it would make sense to automatically apply the limits if the user set a limit, or if they should be forced to pass `True` here. I couldn't decide between it being a useful default ("I set some limit, you should just automatically use them") and having the user be explicit. I don't have a strong opinion either way.

In either case, I think a test of what happens when limits are not specified and `True` is passed here (whether by default or not) would be good (i.e., is an error thrown, or does it just do the same thing as before?).

Reply:
@speth is correct. After manually setting a limit, a user would expect the limit to be applied, which is why I set the default to `True`; the flag allows this to be disabled.

Reply:
Going with "I set some limit, you should just automatically use them", I am leaving the default as `True`, but added a test showing the expected behavior when this is overridden by passing `False`.

Reply:
Is the current behavior retained when no limits are set but `True` is passed? Is there a test for that?

Reply:
The current default behavior is preserved, and there is a test (see `n_baseline` in `test_advance_with_limits` ... the `integrate` function explicitly passes `True` to the `advance` call).

Reply:
Great, thank you!